S3/ConnMan.py

```diff
@@ -88,15 +89,22 @@ def match_hostname_aws(self, cert, e):
         mybucket.example.com.s3.amazonaws.com to be considered a valid
         hostname for the *.s3.amazonaws.com wildcard cert, and for the
         region-specific *.s3-[region].amazonaws.com wildcard cert.
+
+        We also forgive non-S3 wildcard certificates should the
+        hostname match, to allow compatibility with other S3
+        API-compatible storage providers.
         """
-        debug(u'checking SSL subjectAltName against amazonaws.com')
+        debug(u'checking SSL subjectAltName as forgiving wildcard cert')
         san = cert.get('subjectAltName', ())
         for key, value in san:
             if key == 'DNS':
                 if value.startswith('*.s3') and \
                    (value.endswith('.amazonaws.com') and self.hostname.endswith('.amazonaws.com')) or \
                    (value.endswith('.amazonaws.com.cn') and self.hostname.endswith('.amazonaws.com.cn')):
                     return
+                elif value == (Config.host_bucket % {'bucket': '*'}) and \
+                     self.hostname.endswith('.' + '.'.join(Config.host_bucket.split('.')[1:])):
+                    return
```

Comment:
For the "value ==" you take care that the "bucket name" could be anywhere in the url and not necessarily at the left most.
That could be good if you can take care of that for the "self.hostname" check also.
(Note that I don't know if that makes sense relating to certificate dns rules. Like if "fixed_prefix.*.end.of.the.url.com" is allowed.)